fuzz/fuzz_targets/msg_accept_channel_target.rs

```diff
@@ -0,0 +1,137 @@
+extern crate lightning;
+
+use lightning::ln::msgs;
+use lightning::util::reset_rng_state;
+
+use std::sync::atomic::{AtomicUsize,Ordering};
+use lightning::ln::msgs::{MsgEncodable, MsgDecodable};
+
+#[inline]
+pub fn slice_to_be16(v: &[u8]) -> u16 {
+	((v[0] as u16) << 8*1) |
+	((v[1] as u16) << 8*0)
+}
+
+struct InputData<'a> {
+	data: &'a [u8],
+	read_pos: AtomicUsize,
+}
+impl<'a> InputData<'a> {
```

Comment:
Shouldnt need all this complexity for message fuzz targets, check out how peer_crypt_target does it.

Reply:
In channel_target.rs, I saw decode_msg_with_len16() & get_slice_nonadvancing() are used for a message with an optional field so thought it is better for fuzz to converge fast with that way?

Reply:
Ah, sure, yea, but you can also add that without using a reader struct that gets passed around, significantly cutting down on code here.

Reply:
ah, then I don't think I understood correctly how fuzz data feed works. you added Atomic for a multi-threaded test case while there should be no contention for `data` in `fuzz!(|data| {.. }` in a simple function, of course?